---
layout: page
title: Syllabus
permalink: /syllabus/
---
- Day/Time:	Mondays 2:45 - 4:05 and Wednesdays 2:45 - 4:45
- Classroom:	Cuneo Hall 104
- Prerequisites:	COMP 271 and BIOL 388
- Instructor:	 Dr. Heather Wheeler
- email:	<hwheeler1@luc.edu>
- web page:	<http://hwheeler01.github.io/CompBio>
- office hours: Tuesdays 2:30 - 4:30, LSB 324
- TA:


<table style="width: 459px; height: 182px;" class="docutils" border="1">



  <caption><b>Syllabus Index</b></caption> <colgroup><col width="50%"><col width="50%"></colgroup> <tbody valign="top">

    <tr>

      <td><a class="reference internal" href="#objectives">Objectives</a></td>

      <td><a class="reference external" href="#texts">Course Materials</a></td>

    </tr>

    <tr>

      <td><a class="reference external" href="#honesty">Academic Integrity</a></td>

      <td><a class="reference internal" href="#disabilities">Students with Special Needs</a></td>
      
    </tr>

    <tr>

	<td><a class="reference external" href="#assignments">Assignments</a></td>

      <td><a class="reference external" href="#grades">Grades</a></td>

    </tr>

      <tr>

      <td style="vertical-align: top;"><a class="reference external" href="#schedule">Course Schedule</a></td>

      <td style="vertical-align: top;"><a class="reference external" href="#dates">Important Dates</a> </td>

    </tr>

  </tbody>

</table>



<br>



<span style="font-weight: bold;"><a name="objectives"></a>Course Objectives: </span>

- Analyze primary literature in the field of bioinformatics and computational biology- Design and implement advanced data structures for computational biology- Design and implement advanced algorithms for computational biology- Increase proficiency in object-oriented and scripting programming languages


<br>


<span style="font-weight: bold;"><a name="texts"></a>Course Materials: </span>

There is no specific text for this course. Rather, we will be using free online materials and the scientific literature. See <a href="http://hwheeler01.github.io/CompBio/resources/">Resources</a>.

<span style="font-weight: bold;"><a name="honesty"></a>Academic Integrity: </span><br>

Plagiarism and cheating will not be tolerated (see <a href="http://www.luc.edu/academics/catalog/undergrad/reg_academicintegrity.shtml">University Policy</a>) and will be
reported to the college. STUDENTS FOUND TO PRESENT SOMEONE ELSE’S WORK AS THEIR OWN WILL **RECEIVE ZERO POINTS FOR THE ASSIGNMENT**. ANY STUDENT WHO REPEATS SUCH AN ACTION WILL RECEIVE A FAILING GRADE (F) FOR THE COURSE.

The ROSALIND and Python programming assignments at the beginning of the semester can be done independently or in pairs. However, your code must be commented in your own words and turned in independently. ***If you had a partner for an assignment, note his/her name in your code comments***. Cheating include submitting as your own work something that has been written by an outside person (or web site). If you are working on a pair or group project, an “outside person” refers to people other than your partner or team.

After the first month or so, the rest of the course will be spent working on a group project. As part of a group, you are expected to contribute to the same degree as your teammates. As such, “coasting” or letting your teammates carry you is in fact plagiarism; you are taking credit for contributions that you did not make. This is on-par with presenting in a paper someone else’s idea/work as your own without proper citation. Plagiarism and cheating of any form will not be accepted. A student suspected of such behavior will be assessed on an individual basis and action taken per the instructor’s evaluation of the seriousness of thebehavior. Repeat offenders will earn an F for the course.As a contributing member of a group, it is your responsibility to ensure that all members are contributing. Issues that cannot be resolved within the group should be reported to the instructor. (Anonymous reports are welcome.)

<span style="font-weight: bold;"><a name="disabilities"></a>Students with Special Needs:</span>&nbsp; 

Any student with special needs or difficulties in learning and completing course assignments is strongly encouraged to contact the instructor as soon as possible. Please refer to the Student Handbook for student rights and <a href="http://www.luc.edu/sswd/">available resources</a> pertaining to assistance with special needs or disabilities.


<span style="font-weight: bold;"><a name="assignments"></a>Assignments:</span>

Details are <a href="http://hwheeler01.github.io/CompBio/assignments/">here</a>.

<br>


<span style="font-weight: bold;"><a name="grades"></a>Grading:</span><br>

| **Assignment** | **Points**|
| :------------- | ---------:|
| ROSALIND Python Village | 2 |
| ROSALIND Bioinformatics Stronghold 1  | 3 |
| ROSALIND Bioinformatics Stronghold 2 | 5|
| ROSALIND Bioinformatics Armory | 5 |
| Python Parser | 5|
| Initial Group Project Presentation | 10|
| 5-minute Group Progress Presentations (5 points each, 2 total) | 10 |
| Blog Posts (5 points each, 4 total) | 20 |
| Application Note Rough Draft | 5 |
| Final Presentation | 10 |
| Final Project (code) | 15 |
| Final Application Note | 10 |
| **TOTAL** | **100**|





<br>
Course grades are assigned according to the following scale:<br>

| **Grade** | **Points** |
| :--- |:---:|
|A | 90|
|A-| 87|
|B+| 84|
|B | 80|
|B-| 77|
|C+| 74|
|C | 70|
|C-| 67|
|D+| 64|
|D | 60|


<br>



<span style="font-weight: bold;"><a name="schedule"></a>Course Schedule:</span><br>

| **Date**      | **Topics**    | **Exercises** | **DUE by 11:59PM** |
| ------------- |:-------------:| :-----:|---|
|Jan 20|Syllabus, Intro to Python & ROSALIND |INI1-6|<a href="http://rosalind.info/classes/254/">ROSALIND Python Village</a>|
|Jan 25|Sequences as Strings |DNA, REVC, GC, PROT, SUBS, HAMM| |
|Jan 27|Graphs and Motifs |GRPH, LCSM, MPRT |<a href="http://rosalind.info/classes/254/">ROSALIND Bioinformatics Stronghold 1</a> |
|Feb 1|ORFs, Splicing, k-Mers, oh my! | ORF, SPLC, TRAN, LONG, KMER | |
|Feb 3|Constructing a de Bruijn Graph | SETO, DBRU | <a href="http://rosalind.info/classes/254/">ROSALIND Bioinformatics Stronghold 2</a> |
|Feb 8|BioPython | INI, DBPR, FRMT | |
|Feb 10|FASTQ NGS Data|TFSQ, PHRE, FILT, BPHR, BFIL | <a href="http://rosalind.info/classes/254/">ROSALIND Bioinformatics Armory</a> |
|Feb 15|Testing, Debugging | |
|Feb 17|Parsing in Python | VCF to PLINK dosage | VCF parser (via Sakai)|
|Feb 22|Group Project Introduction, form teams | |
|Feb 24|
|Feb 29|
|Mar 2|
|Mar 7| ***No class: SPRING BREAK*** |
|Mar 9| ***No class: SPRING BREAK*** |
|Mar 14|
|Mar 16|
|Mar 21|
|Mar 23|
|Mar 28| ***No class: EASTER BREAK*** |
|Mar 30|
|Apr 4|
|Apr 6|
|Apr 11|
|Apr 13|
|Apr 18|
|Apr 20|
|Apr 25|
|Apr 27|
|May 6| Final Presentation

<br>
*This syllabus is subject to change during the course at the discretion of the instructor.*